src/main.cpp

```diff
@@ -2851,6 +2852,11 @@ bool FindUndoPos(CValidationState &state, int nFile, CDiskBlockPos &pos, unsigne
 
 bool CheckBlockHeader(const CBlockHeader& block, CValidationState& state, bool fCheckPOW)
 {
+    // Check Equihash solution is valid
+    if (fCheckPOW && !CheckEquihashSolution(&block, Params()))
+        return state.DoS(50, error("CheckBlockHeader(): Equihash solution invalid"),
```

Comment:
We could consider this a `100` level DoS instead of `50`, because an incorrect solution has no excuse *ever*.